### PR TITLE
New article: MooseX::Getopt saves Christmas

### PR DESCRIPTION
**Pull request by fleetfootmike**
First pass at a new article - commit and push so someone else knows what's going on, since I did volunteer several months ago. Issue #230

### CODE REVIEW

**Comment by thibaultduponchelle** [comment]
Any issue created for this?

**Comment by thibaultduponchelle** [comment]
Preview:
moosex

File: 2022/incoming/moosex-getopt-saves-christmas.pod [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+    sub run ($self) {
+        # body of script
+    }
+
+=end perl
+
+Santa's right-hand elf's right-hand elf peered curiously over Ernest's shoulder. "Whatcha doing?" she asked.
+
+He sighed. "Fixing the new delivery planning script. I<Someone> commited it without even basic usage docs." Ernest opened a new <delivery_plan> file for the script itself. "Here. Watch.".
```
Comment (thibaultduponchelle) [inline 2022/incoming/moosex-getopt-saves-christmas.pod:57]:
```suggestion
He sighed. "Fixing the new delivery planning script. I<Someone> committed it without even basic usage docs." Ernest opened a new <delivery_plan> file for the script itself. "Here. Watch.".
```

File: 2022/incoming/moosex-getopt-saves-christmas.pod [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+    use feature 'signatures';
+
+    sub run ($self) {
+        # body of script
+    }
+
+=end perl
+
+Santa's right-hand elf's right-hand elf peered curiously over Ernest's shoulder. "Whatcha doing?" she asked.
```
Comment (thibaultduponchelle) [inline 2022/incoming/moosex-getopt-saves-christmas.pod:55]:
```suggestion
Santa's right-hand elf's peered curiously over Ernest's shoulder. "Whatcha doing?" she asked.
```

**Comment by fleetfootmike** [comment]
On 2022-11-28 12:25, Thibault Duponchelle wrote:
> Any issue created for this?

Not yet, trying to slot doing this in in test/swear/edit loops at work 
:D Give me a few mins.

**Review by thibaultduponchelle — CHANGES_REQUESTED** [review]

File: 2022/incoming/moosex-getopt-saves-christmas.pod (hunk 2 above)
Comment (fleetfootmike) [reply 2022/incoming/moosex-getopt-saves-christmas.pod:55]: Er no, Ernest is Santa's right hand elf, she's Ernest's right hand elf, which makes her "Santa's right-hand elf's right-hand elf" :D

**Comment by thibaultduponchelle** [comment]
@oalders I think you're used to MooX::Options that does similar things, I added you as reviewer 🙏🏼 😄

**Comment by fleetfootmike** [comment]
> What do you think about showing the script `--help` output (if not too big)? (not sure if would render well, just throwing the idea/proposal)

I was intending to - expect a commit later today :D

File: 2022/incoming/moosex-getopt-saves-christmas.pod (hunk 2 above)
Comment (thibaultduponchelle) [reply 2022/incoming/moosex-getopt-saves-christmas.pod:55]: Ah OK, sorry sir :D

File: 2022/incoming/moosex-getopt-saves-christmas.pod [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    has 'output' => (
+        is            => 'rw',
+        isa           => 'OutputFormat',
+        required      => 1,
+        documentation => "output format - one of 'plot' or 'slednav'",
+    );
+
+=end perl
+
+He added the rest of the documentation, and another options. "And then... here we go."
```
Comment (oalders) [inline 2022/incoming/moosex-getopt-saves-christmas.pod:133]: Is this meant to be `another option` or `other options`?

File: 2022/incoming/moosex-getopt-saves-christmas.pod [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+Ernest fired off an email.
+
+    Hi,
+
+    When you wake up, please check the commit log for the delivery plan script,
+    and learn from it. The Big Red Boss doesn't have time to read source code
+    to figure out what things we give him are meant to do.
+
+    Also, how on Earth did you managed to come up with an O(N) solution
```
Comment (oalders) [inline 2022/incoming/moosex-getopt-saves-christmas.pod:162]:
```suggestion
    Also, how on Earth did you manage to come up with an O(N) solution
```

**Review by oalders — CHANGES_REQUESTED** [review]
Wonderful stuff. Thank you! Just a couple of comments.

**Comment by thibaultduponchelle** [comment]
@fleetfootmike POD is broken (see my suggestion)

After fixing:
opts

**Comment by fleetfootmike** [comment]
> @fleetfootmike POD is broken (see my suggestion)
> 
> After fixing: opts

Not seeing the suggestion :D

**Review by thibaultduponchelle — CHANGES_REQUESTED** [review]

**Comment by thibaultduponchelle** [comment]
Oh my bad I think I need to "send review" so you can see them

**Review by thibaultduponchelle — APPROVED** [review]

**Comment by thibaultduponchelle** [comment]
Good job @fleetfootmike and thank you a lot. I just approved

**Comment by fleetfootmike** [comment]
On 2022-11-29 10:52, Thibault Duponchelle wrote:
> Good job @fleetfootmike [1] and thank you a lot. I just approved

How short of articles are we? I can chase a few folks up and/or think 
about another one.

**Comment by thibaultduponchelle** [comment]
I think we just hit 25 🎉 so we no longer need any article for 2022( but you can write one now it will go for 2023)

**Review by oalders — APPROVED** [review]

**Pull request merged**